f8a_version_comparator/comparable_version.py

```diff
@@ -0,0 +1,108 @@
+# Copyright © 2018 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Geetika Batra <gbatra@redhat.com>
+#
+
+"""Module to implement Comparable Version class."""
+
+from .item_object import IntegerItem
+from .item_object import StringItem
+from .item_object import ListItem
+
+
+class ComparableVersion():
+    """Class for Comparable Version."""
+
+    def __init__(self, version):
+        """Initialize comparable version class.
+
+        :version: Version supplied as a string
+        """
+        self.parse_version(version)
```

Comment:
It is a good practice to predefine the class attributes in the `__init__` function and just assign the desired values afterwards in the `parse_version`.

Reply:
We would then have to call parse_version explicitly from class object.

Reply:
Oh well, I didn't mean to change the call of `parse_version`, just that it is good practice to declare the class attributes directly in `__init__` function, since it can break the attribute hints in some IDEs later... 

as an Example:
```python
class Foo:
  def __init__(self):
    self.foo_attr = None
    self.parse()
  def parse(self):
    self.foo_attr = "bar"
```

the `foo_attr` is declared directly in the `__init__` instead of in the `parse` method.